addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3606,6 +3606,9 @@ var SnippetsMenu = Widget.extend({
         ];
         loaderContainer.classList.add(...loaderContainerClassList);
         loader.setAttribute('src', '/web/static/img/spin.svg');
+        if (document.querySelector("body").classList.contains("modal-open")) {
+            loaderContainer.classList.add("prevent-spinner");
+        }
```

Comment:
Instead of preventing the loading to appear in the function that creates it, maybe you should directly prevent this function to be called ? :thinking: 
I looked around for a bit and it seems that the loading effect comes from `createLoadingElement` > `_execWithLoadingEffect` > `_onSnippetEditionRequest` > `trigger_up('snippet_edition_request')`  
(in the case where we want to switch theme at least). This `trigger_up` is called by `_onUserValueUpdate` and in this function, you notice a part concerning the confirmation dialog
Maybe you should modify something there and in `_onSnippetEditionRequest` so it calls `_execWithLoadingEffect` with `contentLoading` set to `false` and not always `true`, like a default argument that is `true` when not specified or something like that (not sure if we can do it and if we can do it in stable, maybe ask QSM) :slightly_smiling_face: 

If we can't do that, then maybe prevent the `o_we_ui_loading` class to be added, instead of adding a class that cancels its effects :slightly_smiling_face: 

Also a more general comment, when we create new classes, they should be more specific to website/web_editor, so here, call it `o_we_prevent_spinner` for example :wink:

Reply:
It can be a better solution indeed. I'm gonna investigate. Part of the solution was from qsm or rde (I don't remember) to use a class to hide the spinner.